src/caffe/proto/caffe.proto

```diff
@@ -1235,6 +1235,7 @@ message V1LayerParameter {
     DROPOUT = 6;
     DUMMY_DATA = 32;
     EUCLIDEAN_LOSS = 7;
+    WEIGHTED_EUCLIDEAN_LOSS = 40;
```

Comment:
(1) I think ordering here is done alphabetically; `WEIGHTED_EUCLIDEAN_LOSS` should be roughly at the bottom.
(2) `V1LayerParameter` is only for backward compatibility, I don't think you need to add your layer here anymore.